### PR TITLE
Add squeeze option + test.

### LINKED ISSUE #460: Calling mask.invert_transform(transform(img)) on a 3D image does not return a 3D image.
Calling `mask.invert_transform(transform(img))` on a 3D image appends a 4th dimension.  

To me, this is semantically wrong--if I start with 3D, I should finish with 3D.

The practical problem I hit was while using `nipy`'s plotting functions.  They require 3D data, and this kind of unexpected 4D image made it hard to write generic code there--all other images I was working on were 3D (including others via nilearn functions like `index_img`), but only this one was not.

Here's a script to reproduce the issue.

```python
import nibabel
from nilearn import datasets
from nilearn.image import index_img
from nilearn.input_data import NiftiMasker

haxby_dataset = datasets.fetch_haxby()
mask_filename = haxby_dataset.mask_vt[0]
func_filename = haxby_dataset.func[0]

nifti_masker = NiftiMasker(mask_img=mask_filename, standardize=True)
fmri_img = index_img(nibabel.load(func_filename), 0)  # 3D image
nifti_masker.fit(fmri_img)
print nifti_masker.mask_img_.shape  # 3D mask

fmri_masked = nifti_masker.transform(fmri_img)
fmri_unmasked = nifti_masker.inverse_transform(fmri_masked)
print fmri_img.shape, fmri_unmasked.shape  # 3D start image, 4D end image
```

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Fix #460

### CODE REVIEW

File: nilearn/input_data/base_masker.py [hunk 1]
```diff
@@ -277,15 +278,22 @@ def fit_transform(self, X, y=None, confounds=None, **fit_params):
...
+        if squeeze:
+            img = self._cache(masking.unmask, func_memory_level=1,
+            )(X, self.mask_img_)
+            img = nibabel.funcs.squeeze_image(img)
+        else:            
+            img = self._cache(masking.unmask, func_memory_level=1,
             )(X, self.mask_img_)
         # Be robust again memmapping that will create read-only arrays in
-        # internal structures of the header: remove the memmaped array
+        # internal structures of the header: remove the memmaped array
```
Comment (AlexandreAbraham) [inline nilearn/input_data/base_masker.py]: Trailing spaces are not PEP8. You should add automatic check of PEP8 in your editor.

File: nilearn/input_data/base_masker.py [hunk 2]
```diff
@@ -277,15 +278,22 @@ def fit_transform(self, X, y=None, confounds=None, **fit_params):
...
+            img = nibabel.funcs.squeeze_image(img)
+        else:            
+            img = self._cache(masking.unmask, func_memory_level=1,
             )(X, self.mask_img_)
         # Be robust again memmapping that will create read-only arrays in
-        # internal structures of the header: remove the memmaped array
+        # internal structures of the header: remove the memmaped array                      
         try:
-            img._header._structarr = np.array(img._header._structarr).copy()
+              img._header._structarr = np.array(img._header._structarr).copy()
```
Comment (AlexandreAbraham) [inline nilearn/input_data/base_masker.py]: Not PEP8.

File: nilearn/input_data/base_masker.py [hunk 3]
```diff
@@ -277,15 +278,22 @@ def fit_transform(self, X, y=None, confounds=None, **fit_params):
                               ' will be used.' % self.__class__.__name__)
                 return self.fit(**fit_params).transform(X, confounds=confounds)
 
-    def inverse_transform(self, X):
-        img = self._cache(masking.unmask, func_memory_level=1,
+    def inverse_transform(self, X, squeeze=False):
+        # Select squeeze as True if needed to return an image of squeezing with 
+        # last dimension "1" from a 4D format to 3D image format
+        if squeeze:
+            img = self._cache(masking.unmask, func_memory_level=1,
```
Comment (kshmelkov) [inline nilearn/input_data/base_masker.py]: May be put it outside "if" clause as it repeats in both branches?

File: nilearn/input_data/base_masker.py [hunk 4]
```diff
@@ -277,15 +278,22 @@ def fit_transform(self, X, y=None, confounds=None, **fit_params):
...
+            )(X, self.mask_img_)
```
Comment (AlexandreAbraham) [inline nilearn/input_data/base_masker.py]: Why is this line in the condition? Since it is always done, it should be above.

**Comment by GaelVaroquaux** [comment]
Hadn't we decided that our plotting should support 4D and that we should forget about the squeeze idea.

File: nilearn/input_data/tests/test_nifti_masker.py [hunk 5]
```diff
@@ -275,3 +275,21 @@ def test_compute_epi_mask():
 
     assert_false(np.allclose(mask1.get_data(),
                              mask4.get_data()[3:12, 3:12]))
+
+
+def test_squeeze():
+    mask = np.ones((10, 10, 10))
+    mask_img = Nifti1Image(mask, np.eye(4))
+
+    data = Nifti1Image(np.random.random((10, 10, 10, 1)), np.eye(4))\
```
Comment (lesteve) [inline nilearn/input_data/tests/test_nifti_masker.py]: unnecessary backslash at the end of the line?

**Comment by AlexandreAbraham** [comment]
Closing since we fixed that by making plotting function accept 4d image.

File: nilearn/input_data/base_masker.py [hunk 6]
```diff
@@ -277,15 +278,22 @@ def fit_transform(self, X, y=None, confounds=None, **fit_params):
                               ' will be used.' % self.__class__.__name__)
                 return self.fit(**fit_params).transform(X, confounds=confounds)
 
-    def inverse_transform(self, X):
-        img = self._cache(masking.unmask, func_memory_level=1,
+    def inverse_transform(self, X, squeeze=False):
+        # Select squeeze as True if needed to return an image of squeezing with
```
Comment (eickenberg) [inline nilearn/input_data/base_masker.py]: This could be made into a docstring

**Comment by eickenberg** [comment]
Oops, guess I'm late to the game, never mind.
